python/tvm/driver/tvmc/compiler.py

```diff
@@ -196,9 +196,9 @@ def compile_model(
     for codegen_from_cli in extra_targets:
         codegen = composite_target.get_codegen_by_target(codegen_from_cli["name"])
         partition_function = codegen["pass_pipeline"]
-        mod = partition_function(mod, params)
+        mod, codegen_config = partition_function(mod, params)
         if codegen["config_key"] is not None:
-            config[codegen["config_key"]] = codegen_from_cli["opts"]
+            config[codegen["config_key"]] = codegen_from_cli["opts"] or codegen_config
```

Comment:
It means the generated codegen_config will be completely overwritten by the user config, but we should make it like
```python
codegen_config.update(codegen_from_cli["opts"] if codegen_from_cli["opts"] is not None else {})
```

Reply:
The config returned by `partiton_for_tensorrt` reflects the user config by the`codegen_from_cli` argument.
And, the other composite_targets don't return config.
 So, config with a `extra_codegen` can be overwritten codegen_config if any.

Reply:
> The config returned by partiton_for_tensorrt reflects the user config by the codegen_from_cli argument.

This is not guaranteed. What if `partition_for_xxx` only uses a part of the configs in `codegen_from_cli`?

Reply:
Sorry, I don't know it means.

The implementation before this change use all of `codegen_from_cli` as the config.
If should consider only uses a part of `partiton_for_xxx`, it's due to not this change but original implementation.
Or, should it apply also in this change?

Reply:
Sorry I misinterpreted the meaning of `codegen_from_cli` (their names are sort of confusing...).
Since `codegen_from_cli` is given by each composite target, this way should be fine.